### PR TITLE
Add FXIOS-4075 [v113] A confirmation menu/undo option should be displayed when tapping the "Close All Inactive Tabs" button

### PR DESCRIPTION
**Pull request by yoanarios**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-4075)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/10439)

### Description
- Create InactiveTabs folder
- Add Toast to undo Inactive tabs close using existing CFR delegate
- Move UX struct in ViewController
- Add unit test 

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [x] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by yoanarios** [comment]
@OrlaM tagging you for review for possible conflict with the refactoring

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Print() function seems to be used in file Client/Frontend/Browser/GridTabViewController.swift at line 528. Please remove this usage from production code or use BrowserKit Logger.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 29.08%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 12 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 26.7
| File | Coverage ||
| --- | --- | --- |
TabCell.swift | 78.04% | ✅
InactiveTabCell.swift | 0.0% | ⚠️
TabTrayViewController.swift | 48.05% | ⚠️
InactiveTabItemCell.swift | 0.0% | ⚠️
GroupedTabCell.swift | 0.0% | ⚠️
InactiveTabItemCellModel.swift | 0.0% | ⚠️
InactiveTabHeader.swift | 0.0% | ⚠️
InactiveTabViewModel.swift | 86.96% | ✅
GridTabViewController.swift | 36.88% | ⚠️
TabDisplayManager.swift | 50.31% | ✅

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 8598657d01b0378f34d88ed7b87dbb3ae7ddfcbc
</p>

File: Client/Frontend/Browser/TabDisplayManager.swift [hunk 1]
```diff
@@ -630,6 +630,43 @@ extension TabDisplayManager: GroupedTabDelegate {
 
 // MARK: - InactiveTabsDelegate
 extension TabDisplayManager: InactiveTabsDelegate {
+    func shouldCloseInactiveTab(tab: Tab) {
+        removeInactiveTabAndReloadView(tabs: [tab])
+        TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .inactiveTabTray, value: .inactiveTabSwipeClose, extras: nil)
```
Comment (nbhasin2) [inline Client/Frontend/Browser/TabDisplayManager.swift:635]: nit: indent
```suggestion
        TelemetryWrapper.recordEvent(category: .action,
                                     method: .tap,
                                     object: .inactiveTabTray,
                                     value: .inactiveTabSwipeClose,
                                     extras: nil)
```

**Review by nbhasin2 — APPROVED** [review]
I tested on both iPad and iPhone and works well 👍🏼 

Note: If you close the app while user taps on close all tabs, then that won't actually close any tabs. I am not sure what we can do about it considering there is a delay attached to the snackbar. IMO this is fine!!

**Pull request merged**